ddns.php

```diff
@@ -4,94 +4,119 @@
 require __DIR__ . '/Cloudflare.php';
 
 $confFile = __DIR__ . '/config.php';
-if (!file_exists($confFile))
-{
-  echo "Missing config file. Please copy config.php.skel to config.php and fill out the values therein.\n";
-  return 1;
+if (!file_exists($confFile)) {
+    echo "Missing config file. Please copy config.php.skel to config.php and fill out the values therein.\n";
+    return 1;
 }
 
 $config = require $confFile;
 
-foreach (['cloudflare_email', 'cloudflare_api_key', 'domain', 'record_name', 'ttl', 'protocol'] as $key)
-{
-  if (!isset($config[$key]) || $config[$key] === '')
-  {
-    echo "config.php is missing the '$key' config value\n";
-    return 1;
-  }
+foreach ([
+             'cloudflare_email',
+             'cloudflare_api_key',
+             'domain',
+             'record_name',
+             'ttl',
+             'protocol',
+         ] as $key) {
+    if (!isset($config[$key]) || $config[$key] === '') {
+        echo "config.php is missing the '$key' config value\n";
+        return 1;
+    }
 }
 
 $api = new Cloudflare($config['cloudflare_email'], $config['cloudflare_api_key']);
 
-$domain     = $config['domain'];
-$recordName = $config['record_name'];
-
-if (isset($config['auth_token']) && $config['auth_token'])
-{
-  // API mode. Use IP from request params.
-  if (empty($_GET['auth_token']) || empty($_GET['ip']) || $_GET['auth_token'] != $config['auth_token'])
-  {
-    echo "Missing or invalid 'auth_token' param, or missing 'ip' param\n";
-    return 1;
-  }
-  $ip = $_GET['ip'];
+// default to first value of config array
+$domain = $config['domain'][0];
+$recordName = $config['record_name'][0];
+// set domain and record from request if value exists in config
+if ($_GET['domain'] || $_GET['record']) {
+    if (
+        $_GET['domain'] &&
+        $_GET['record'] &&
```

Comment:
this looks like you'll get an ` Undefined index: record` warning if the `domain` param is passed but the `record` param is not

Reply:
I don't think so - the inner `if` checks if both are there. but maybe I'm wrong there - I'm happy to change things around if it's needed.

Reply:
its ok i fixed it up. thanks